beacon-chain/blockchain/service.go

```diff
@@ -215,7 +184,8 @@ func (c *ChainService) updateHead(slot uint64) {
 	// We fire events that notify listeners of a new block (or crystallized state in
 	// the case of a state transition). This is useful for the beacon node's gRPC
 	// server to stream these events to beacon clients.
-	transition := c.chain.IsCycleTransition(slot)
+	cState := c.chain.CrystallizedState()
+	transition := cState.IsCycleTransition(c.candidateBlock.SlotNumber())
```

Comment:
Do you need `transition`? or can you inline this method call with the if statement?